driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -129,6 +125,14 @@ void cancel() {
     cancelPendingExecutions(null);
   }
 
+  String getId() {
+    // atomicity is not required to set the value
```

Comment:
This method could be called concurrently from different speculative executions. Are you saying atomicity is not required because even if 2 threads enter the method at the same time, the effect is going to be deterministic and idempotent, i.e. field `id` will be eventually set to `Long.toString(System.identityHashCode(this))`?

Reply:
yes, you are right, the result is deterministic and double evaluation is not a problem if it will happen (an atomic construction, like AtomicLong, would be more expensive here)